core/ops.rs

```diff
@@ -25,17 +27,18 @@ pub type CoreError = ();
 pub type CoreOp = Op<CoreError>;
 
 /// Main type describing op
-type OpDispatcher = dyn Fn(&[u8], Option<PinnedBuf>) -> CoreOp;
+type OpDispatcher =
+  dyn Fn(&[u8], Option<PinnedBuf>) -> CoreOp + Send + Sync + 'static;
 
 #[derive(Default)]
 pub struct OpRegistry {
-  dispatchers: Vec<Box<OpDispatcher>>,
-  name_to_id: HashMap<String, OpId>,
+  dispatchers: RwLock<Vec<Arc<Box<OpDispatcher>>>>,
+  name_to_id: RwLock<HashMap<String, OpId>>,
```

Comment:
😭 this might impact performance, hot path will require to acquire a read lock everytime

Reply:
It doesn't look like it did in benchmarks.

Reply:
Yes, but they're flaky... Anyway why do you need to put `dispatchers` behind the lock? In native plugin PR you already have `Arc<Mutex<Isolate>>` to you should be able to register new ops quite easily

Reply:
In most cases during dispatch we already have a lock on isolate somewhere else.

Reply:
I see... I'm curious to see it combined with native plugins PR